journals/apps/journals/views.py

```diff
@@ -84,3 +97,106 @@ def get(self, request, *args, **kwargs):
                 'collections': collections,
                 'current_collection': current_collection,
             })
+
+
+class JournalAccessView(View):
+    """
+    View to give journal's access to users.
+    """
+    template = "admin/journals/journal/import_users.html"
+
+    def get_context(self, request, journal_id, journal_form=None):
+        """
+        Returns the context for UsersJournalAccessForm.
+        """
+        journal = Journal.get_journal_by_id(journal_id)
+        context = {
+            'url': reverse('admin:import_users', args=[journal_id]),
+            'journal_access_form': journal_form if journal_form else UsersJournalAccessForm(
+                initial={'expiration_date': get_default_expiration_date(journal)}
+            ),
+            'opts': journal._meta,
+        }
+        context.update(admin.site.each_context(request))
+        return context
+
+    def get_or_create_journal_users(self, request, usernames):
+        """
+        Checks whether the usernames are valid (do exist in journal database)
+        if a username doesn't exist then fetch its data from LMS and create
+        a user in journal.apps.models.USER
+
+        Returns:
+            returns the set of valid usernames (exist in journal database)
+        """
+        all_users = User.objects.all()
+        non_existing_users = [username for username in usernames if not all_users.filter(username=username).exists()]
```

Comment:
To minimize DB calls, you can flip this to getting valid users, then diffing for the non_existing_users:
```
valid_usernames = all_users.filter(username__in=usernames).values_list('username', flat=True)
non_existing_users = set(usernames) - set(existing_users)
```

Reply:
@tuchfarber both cases will have the same number of queries. But your method will save the iteration of the loop. Thanks